File: Software-Praktikum/src/server/bo/UserBO.py
from .Businessobject import Businessobject


class User (Businessobject):

    def __init__(self):
        super().__init__()
        self.name = ""  # Der Name des Benutzers.
        self.email = ""  # Die E-Mail-Adresse des Benutzers.
        self.user_id = ""  # Die extern verwaltete User ID.

    def get_name(self):
        """Auslesen des Benutzernamens."""
        return self.name

    def set_name(self, value):
        """Setzen des Benutzernamens."""
        self.name = value

    def get_email(self):
        """Auslesen der E-Mail-Adresse."""
        return self.email

    def set_email(self, value):
        """Setzen der E-Mail-Adresse."""
        self.email = value

    def get_user_id(self):
        """Auslesen der externen User ID (z.B. Google ID)."""
        return self.user_id

    def set_user_id(self, value):
        """Setzen der externen User ID (z.B. Google ID)."""
        self.user_id = value

    def __str__(self):
        """Erzeugen einer einfachen textuellen Darstellung der jeweiligen Instanz."""
        return "User: {}, {}, {}, {}".format(self.get_id(), self.name, self.email, self.user_id)

    @staticmethod
    def from_dict(dictionary=dict()):
        """Umwandeln eines Python dict() in einen User()."""
        obj = User()
        obj.set_id(dictionary["id"])  # eigentlich Teil von BusinessObject !
        obj.set_name(dictionary["name"])
        obj.set_email(dictionary["email"])
        obj.set_user_id(dictionary["user_id"])
        return obj

File: Software-Praktikum/Frontend/lernapp/src/API/UserBO.js
import BusinessObject from "./BusinessObject.js"

/**
 * @author [Mihriban Dogan](https://github.com/mihriban-dogan)
 */

export default class UserBO extends BusinessObject{
    /**
   * @param {*} name- Name des Users
   * @param {*} email - Google-Email des Users
   * @param {*} user_id- die User ID
   */

    constructor(name, email, user_id) {
        super();
        this.name = name
        this.email = email
        this.user_id = user_id

    }
    //Hier folgen die Setter und Getter der einzelnen Parameter

    setName(name) {
        this.name = name;
    }

    getName() {
        return this.name;
    }

    setEmail (email) {
        this.email = email;
    }

    getEmail() {
        return this.email;
    }

    setUser_id (user_id) {
        this.user_id = user_id;
    }

    getUser_id() {
        return this.user_id;
    }


    static fromJSON(user) {
        let result = [];

        if (Array.isArray(user)) {
            user.forEach((c) => {
                Object.setPrototypeOf(c, UserBO.prototype);
                result.push(c);
            })
        } else {
            let c = user;
            Object.setPrototypeOf(c, UserBO.prototype);
            result.push(c);
        }

        return result;
    }
}
